addons/account/models/account_move.py

```diff
@@ -3498,6 +3498,10 @@ def _post(self, soft=True):
                 customer_count[invoice.partner_id] += 1
             elif invoice.is_purchase_document():
                 supplier_count[invoice.partner_id] += 1
+            elif invoice.move_type == 'entry':
+                sale_amls = invoice.line_ids.filtered(lambda line: line.balance != 0 and bool(line.partner_id) and line.account_id.account_type == 'asset_receivable')
```

Comment:
```suggestion
                sale_amls = invoice.line_ids.filtered(lambda line: line.debit and line.partner_id and line.account_id.account_type == 'asset_receivable')
```
- `line.partner_id` is either False or a Recordset<res.partner> (which is truthy)
- A line in a receivable account represents a partner's debt if it's a debit. If it's a credit, then it means that we owe the partner. In this situation, I think it makes sense to count the misc entry only if the partner owes us.